### PR TITLE
Big ol nightly refactor

### PR DESCRIPTION
**Pull request by ajpal**
Make the data flatt

### CODE REVIEW

File: infra/nightly-resources/handlers.js [hunk 1]
```diff
@@ -74,18 +78,8 @@ function toggleCheckbox(mode, set) {
 }
 
 async function loadBaseline(url) {
-  const data = await getBench(url + "/");
   clearWarnings();
-  GLOBAL_DATA.baselineRun = data;
-  const benchmarkNames = Object.keys(GLOBAL_DATA.currentRun);
-  // Add warnings if the baseline run had a benchmark that the current run doesn't
```
Comment (oflatt) [inline infra/nightly-resources/handlers.js:81]: I think this warning got dropped in this PR

File: infra/nightly-resources/index.js [hunk 2]
```diff
@@ -30,254 +30,22 @@ function clearWarnings() {
...
-  parsed.sort((l, r) => {
-    if (l.name < r.name) {
-      return -1;
-    }
-    if (l.name > r.name) {
-      return 1;
-    }
-    return 0;
+  const byBench = {};
+  Array.from(GLOBAL_DATA.enabledBenchmarks).forEach((benchmark) => {
```
Comment (oflatt) [inline infra/nightly-resources/index.js:39]: Why do we need `Array.from`?

**Review by oflatt — CHANGES_REQUESTED** [review]
Looks good!
However, running it locally it looks like the links to llvm files are broken

File: infra/nightly-resources/index.js (hunk 2 above)
Comment (ajpal) [reply infra/nightly-resources/index.js:39]: It's a `Set`

Comment (ajpal) [reply infra/nightly-resources/index.js]: Actually, not needed, because order of iteration won't matter here.

**Review by oflatt — APPROVED** [review]
Great!

**Pull request merged**